pymongo/mongo_client.py

```diff
@@ -2273,6 +2273,7 @@ def handle(
     ) -> None:
         if self.handled or exc_type is None:
             return
+        assert exc_val is not None
```

Comment:
I suggest switching the order of the conditions in the if statement instead of adding an assert.

Reply:
sounds good